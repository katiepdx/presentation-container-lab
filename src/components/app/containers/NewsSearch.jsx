// holds the state for search, loading, and article state
// Needs a behavior test

import React, { Component } from 'react'
import { getNewsArticles } from '../services/news-api'
import Loading from '../presentational/Loading'
import Search from '../presentational/Search'
import ArticlesList from '../presentational/ArticlesList'

export default class NewsSearch extends Component {
  // set state 
  state = {
    loading: false,
    search: '',
    foundArticles: []
  }

    handleChange = (e) => {
      this.setState({ search: e.target.value })
    }

    handleClick = async (e) => {
      e.preventDefault();

      // conditional
      if(!this.state.search) return alert('Please enter a search')

      // set loading to true to get loading spinner
      this.setState({ loading: true });

      // fetch from api using user's search
      const foundArticles = await getNewsArticles(this.state.search);

      // update state with user's search
      this.setState({ foundArticles: foundArticles, loading: false })
    }

  render() {
        // deconstruct state 
        const { loading, foundArticles } = this.state;

        // conditional rendering 
        if(loading) return <Loading />
    
        // Use array from state
        return (
          <div data-testid="display">
            <h1>Welcome!</h1>
            <Search onChange={this.handleChange} onClick={this.handleClick} />
            <ArticlesList foundArticles={foundArticles}/>
          </div>
        )
      }
  }

