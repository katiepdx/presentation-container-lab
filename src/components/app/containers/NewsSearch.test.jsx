// behavior test 
import React from 'react'
import { fireEvent, render, screen } from '@testing-library/react'
import NewsSearch from './NewsSearch'

describe('NewsSearch container', () => {
  afterAll(() => cleanup());

  it('updates search field value to tree', () => {
    // render NewsSearch page 
    render(<NewsSearch />)

    // get the search input field
    const searchInput = screen.getByTestId("search-input")

    // select display
    const display = screen.getByTestId("display")

    // use the search term - tree
    fireEvent.change(searchInput, {
      target: { value: 'tree' }
    })
    
    console.log('SEARCH INPUT VALUE', searchInput.value)

    expect(searchInput).toHaveValue('tree')
    expect(searchInput).toMatchSnapshot()
  })
})
