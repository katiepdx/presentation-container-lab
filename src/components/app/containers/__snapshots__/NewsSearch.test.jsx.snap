// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsSearch container updates search field value to tree 1`] = `
<input
  data-testid="search-input"
  id="text"
  name="text"
  type="text"
  value=""
/>
`;
